pkg/envoy/cds/response.go

```diff
@@ -25,32 +25,24 @@ func NewResponse(meshCatalog catalog.MeshCataloger, proxy *envoy.Proxy, _ *xds_d
 		return nil, err
 	}
 
-	if proxy.Kind() == envoy.KindGateway {
-		// Build remote clusters based on allowed outbound services
-		for _, dstService := range meshCatalog.ListOutboundServicesForIdentity(proxyIdentity) {
-			cluster, err := getUpstreamServiceCluster(proxyIdentity, dstService, cfg)
+	if proxy.Kind() == envoy.KindGateway && cfg.GetFeatureFlags().EnableMulticlusterMode {
+		for _, dstService := range meshCatalog.ListOutboundServicesForGateway() {
+			kubeSvc := meshCatalog.GetKubeController().GetService(dstService)
+			cluster, err := getGatewayUpstreamServiceCluster(dstService, cfg, kubeSvc.Spec.Ports)
```

Comment:
I suggest not leak k8s here. This is going to come back and bite us hard. Instead, define a method in `catalog` that returns a collection of all configuration required by XDS to configure the mutlicluster gateway, much like `trafficpolicy.EgressTrafficPolicy` and `trafficpolicy.IngressTrafficPolicy`. Within that catalog function, you could set properties in way that works for this use case.

Reply:
I have updated the logic to not leak k8s, take a look and let me know it the implementation works